src/io/multi_writer.cr

```diff
@@ -0,0 +1,55 @@
+module IO
+  # An IO which writes to a number of underlying writer IOs.
+  #
+  # ```
+  # io1 = IO::Memory.new
+  # io2 = IO::Memory.new
+  # writer = IO::MultiWriter.new(io1, io2)
+  # writer.puts "foo bar"
+  # io1.to_s # => "foo bar\n"
+  # io2.to_s # => "foo bar\n"
+  # ```
+  class MultiWriter
+    include IO
+
+    # If `sync_close` is true, closing this IO will close all of the underlying
+    # IOs.
+    property? sync_close
+    getter? closed = false
+
+    @writers : Array(IO)
+
+    # Creates a new `IO::MultiWriter` which writes to *writers*. If
+    # *sync_close* is set, calling `#close` calls `#close` on all underlying
+    # writers.
+    def initialize(@writers : Array(IO), @sync_close = false)
+    end
+
+    # Creates a new `IO::MultiWriter` which writes to *writers*. If
+    # *sync_close* is set, calling `#close` calls `#close` on all underlying
+    # writers.
+    def initialize(*writers : IO, @sync_close = false)
+      @writers = writers.map(&.as(IO)).to_a
+    end
+
+    def write(slice : Slice(UInt8))
+      check_open
+
+      @writers.each { |writer| writer.write(slice) }
+    rescue ex
+      close
```

Comment:
Here's a reason for closing: if one of the write calls fails, then one of the underlying IOs may have written more than the rest. This means that the IOs are no longer consistent, and so we should fail all subsequent calls so that the two IOs don't go out of sync. 

Unfortunately this isn't desired behaviour for logging because you don't want stdout to stop when the disk is full, so maybe it should be configurable?